modules/code-editor/src/views/full/SidePanel.tsx

```diff
@@ -225,50 +222,57 @@ class PanelContent extends React.Component<Props> {
     )
   }
 
-  _buildHooksActions() {
+  _buildHooksActions(showGlobalHooks: boolean) {
     const hooks = Object.keys(HOOK_SIGNATURES).map(hookType => ({
       id: hookType,
       label: hookType
         .split('_')
         .map(x => x.charAt(0).toUpperCase() + x.slice(1))
         .join(' '),
-      onClick: () => this.props.createFilePrompt('hook', true, hookType)
+      onClick: () => this.createFilePrompt('hook', hookType)
     }))
 
+    const items = [
+      {
+        label: 'Event Hooks',
+        items: hooks.filter(x =>
+          [
+            'before_incoming_middleware',
+            'after_incoming_middleware',
+            'before_outgoing_middleware',
+            'after_event_processed',
+            'before_suggestions_election',
+            'before_session_timeout'
+          ].includes(x.id)
+        )
+      },
+      {
+        label: 'Bot Hooks',
+        items: hooks.filter(x => ['after_bot_mount', 'after_bot_unmount', 'before_bot_import'].includes(x.id))
+      }
+    ]
+
+    if (showGlobalHooks) {
+      items.push(
+        {
+          label: 'General Hooks',
+          items: hooks.filter(x => ['after_server_start'].includes(x.id))
+        },
+        {
+          label: 'Pipeline Hooks',
+          items: hooks.filter(x =>
+            ['on_incident_status_changed', 'on_stage_request', 'after_stage_changed'].includes(x.id)
```

Comment:
is  `on_incident_status_changed` really a pipeline hook ? To me, this would be mode of a genera hook. One could act on incident status changed without having configured a pipeline.

Reply:
Good point, I was always wondering what it was doing there